pennylane/qaoa/cost.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This file contains functions that generate QAOA cost Hamiltonians corresponding to
+different optimization problems.
+"""
+
+import networkx as nx
+import pennylane as qml
+
+
+def MaxCut(graph):
+    r"""Returns the QAOA cost Hamiltonian corresponding to the MaxCut problem, for a given graph.
+
+    The goal of the MaxCut problem for a particular graph is to find a partition of nodes into two sets,
+    such that the number of edges in the graph with endpoints in different sets is maximized. Formally,
+    we wish to find the `cut of the graph <https://en.wikipedia.org/wiki/Cut_(graph_theory)>`__ such
+    that the number of edges crossing the cut is maximized.
+
+    The MaxCut Hamiltonian is defined as:
+
+    .. math:: H_C \ = \ \frac{1}{2} \displaystyle\sum_{(i, j) \in E(G)} Z_i Z_j \ - \ \mathbb{I}
+
+    where :math:`G` is some graph and :math:`Z_i` and :math:`Z_j` are the Pauli-Z operators on the
+    :math:`i`-th and :math:`j`-th wire respectively.
+
+    *Recommended mixer Hamiltonian:* :func:`~.qaoa.x_mixer`
+
+    *Recommended initialization circuit:* Even superposition over all basis states
```

Comment:
Nice!

Some ideas on formatting:
Could make them bold and with the following text being a new line, which might fit more into the style
![image](https://user-images.githubusercontent.com/49409390/89047874-0afbae00-d31d-11ea-916c-8d3a6fb1d963.png)

Could also place in a `.. note::` box to make it very noticeable

Reply:
Sounds good!